strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/LoggingManagementController.java

```diff
@@ -238,11 +237,111 @@ public ResponseEntity uploadLogbackConfiguration(HttpServletRequest request,
         catch (IOException | LoggingConfigurationException e)
         {
             String message = "Could not upload logback configuration.";
+            
             logger.error(message, e);
 
-            return ResponseEntity.status(BAD_REQUEST)
-                                 .body(getResponseEntityBody(message, accept));
+            return ResponseEntity.status(BAD_REQUEST).body(getResponseEntityBody(message, accept));
         }
     }
 
+    @ApiOperation(value = "Used to get log directory.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The log directory was retrieved successfully."),
+                            @ApiResponse(code = 400, message = "Could not download log directory.") })
+    @GetMapping(value = { "/logs",
+                          "/logs/**" },
+                produces = TEXT_PLAIN_VALUE)
+    public void generateLogDirectoryListing(HttpServletRequest request,
+                                            HttpServletResponse response)
+            throws IOException
+    {
+        String requestUriString = request.getRequestURI();
+        String uriLogDirPath = requestUriString.replace("/logging", "");
+        Path localLogDirPath = Paths.get(PropertyUtils.getVaultDirectory(), uriLogDirPath);
+        
+        if (Files.notExists(localLogDirPath))
+        {
+            response.sendError(404, "File not found. Variables: " + uriLogDirPath + " " + localLogDirPath.toString());
```

Comment:
You need to add `return;` in this block. If you don't - the code will send the response to the client and continue executing the rest of the code.

Reply:
Oh yes good catch!